gapidapk/deviceinfo.go

```diff
@@ -106,47 +106,26 @@ func fetchDeviceInfo(ctx context.Context, d adb.Device) error {
 
 	var cleanup app.Cleanup
 
-	packages := []string{}
-	driver, err := d.GraphicsDriver(ctx)
+	supported, packageName, nextCleanup, err := d.PrepareGpuProfiling(ctx, apk.InstalledPackage)
+	cleanup = cleanup.Then(nextCleanup)
 	if err != nil {
-		// If there's an error, keep going to attempt to use GPU profiling
-		// libraries in system image.
-		log.W(ctx, "Failed to query developer driver: %v, assuming no developer driver found.", err)
+		cleanup.Invoke(ctx)
+		return err
 	}
-	if driver.Package != "" {
-		log.I(ctx, "Using GPU profiling libraries from developer driver package: %v.", driver.Package)
-
-		// Set up device info service to use prerelease driver.
-		nextCleanup, err := adb.SetupPrereleaseDriver(ctx, d, apk.InstalledPackage)
+	if supported {
+		// Set driver package
+		nextCleanup, err := android.SetupLayers(ctx, d, apk.Name, []string{packageName}, []string{})
```

Comment:
We used to check that `packageName` was not empty before calling `SetupLayers`. You should do that, too, or document in the commit message why you are changing this functionality in a refactor.

Reply:
See other conversation.